### ISSUE TITLE
Melting pot ;-)

### ISSUE THREAD

**Issue by Sami32** [DigitalMediaServer/DigitalMediaServer#5]
- I think that "Engines are in descending order..." should be better displayed as a popup while crossing the entry over the engines windows ?

-  Unusual warnings are throwed in the log at starting time concerning the renderers:
    WARN 2017-08-01 00:08:08.265 [AWT-EventQueue-0] Renderer Yamaha RX-V671 found twice, ignoring repeated entry

**EDIT**:

- I also saw that, same for UMS, i just verified, Streamium renderer should have been grouped under "Philips" folder and not leaved alone.

**Comment by Sami32** [comment]
I don't really know where to post it, so here i go.

We can restart the server every time we want, also if nothing has been changed.
Is that was expected ?

**Comment by Nadahar** [comment]
@Sami32 Yes, you can in UMS as well, you just don't see anything happening with the "dead" button. Over time I'm planning to change that so that you'll never (or almost never) need to "restart the server". I'll just have to find new use for my button 😢

**Comment by Sami32** [comment]
What about free plans like these in an eventual future ? I think that @taconaut is using its own server.
- https://hub.openshift.com/quickstarts/98-jenkins-server, https://jenkins.io/solutions/github/, https://ci.jenkins.io/
- https://circleci.com/pricing/#build-linux
- https://www.phacility.com/pricing/
- https://www.shippable.com/pricing.html
- https://www.bitrise.io/pricing
Some comparaisons are done [there](http://mjdinteractive.com/picking-a-continuous-integration-platform-for-mobile-projects/).

**Comment by Sami32** [comment]
The search button in _Navigation/Share settings_ should be from a different color, IMO, as i found it confusing with the 2 first buttons adding and deleting folders.
That could only me though.

**Comment by Nadahar** [comment]
@Sami32 What exactly are you looking for? Travis-CI already does everything we need except Windows builds. I've looked for free, online Windows builds before, but didn't find much that seemed like it was worth it. I guess I could always install a Jenkins instance on a VM, but all those builds use a *lot* of bandwidth since everything is downloaded over and over and over again.

**Comment by Sami32** [comment]
Yes, it's why i checked at these free plans. In the purpose of doing Travis-CI or like, without using our personal bandwidth and for free ;-)

**Comment by Nadahar** [comment]
But, Travis-CI is also free, and doesn't have a lot of the limitations the other projects have. So what's the advantage?

**Comment by Sami32** [comment]
The choice and the bandwidth available.
Do you have a better idea for a free automatic builds usable by users, not using our bandwdith ?

**Comment by Nadahar** [comment]
@Sami32 Travis can also be used to upload builds, we just don't do it because there's no need. You'd still need somewhere to host them, I don't think the CIs offer storage space. Rented storage space online is very expensive, and when each build is ~250 MB (for 3 platforms) the GB's go quickly. Also, what would be the use? Most of the time, if anyone needs to test something, they want a branch/PR, not master. Did you want to upload all the branch builds as well?

**Comment by Nadahar** [comment]
@Sami32 
> The search button in Navigation/Share settings should be from a different color, IMO, as i found it confusing with the 2 first buttons adding and deleting folders.

I really don't want to change the color of the scan button, as it's animated and takes many hours (don't remember, but around 12 hours on my computer I think) to render. Changing the color of the add/remove buttons would be better then, since they aren't animated - but what color shoud they have?

**Comment by Sami32** [comment]
I just want to give the possibility to the users to be able to what they do with the @taconaut Jenkins server. It seem that i was not clear as i thought, as usual.

**Comment by Nadahar** [comment]
@taconaut's Jenkins server use his personal bandwidth, and isn't exactly a "public service" 😉 I hope not many users use it, I don't think too many people know about it.

**Comment by Sami32** [comment]
Damned ! I was far to imagine that.
I was thinking of green or orange, as you already used them, or yellow.
The effect of the scan is not too obvious to me, in fact i missed it at first and searched to see if the scan was finished, but i think now that perhaps because of my "screen protection" with its red filter that affect it.
Forget about, now that you told me that i don't think its worst that work. Sorry i didn't know that was a such work !!!

**Comment by Sami32** [comment]
Yes, i told you that is using its own server.
I don't know how many users use it, but i learned from it because @taconaut  posted it on the forum some years ago.

**Comment by Nadahar** [comment]
@Sami32 I think it's quite easy to see that the scan is running, I'm guessing your "screen protection" has to do with it, along with your tiny monitor.

The 12 hours is just for the computer to render all the images of the animation, let's not mention how many hours it took me to actually *make* it 😉

**Comment by Sami32** [comment]
Yeah, better not :wink:

**Comment by Nadahar** [comment]
@Sami32 You know my "perfectionism"? This is the actual quality I made all the buttons in:

![button-scan-busyf14_full](https://user-images.githubusercontent.com/12900583/28852081-e708ed96-7726-11e7-94f0-ecba1f3d1b92.png)

The scan animation consists of 28 frames * 2 (28 frames with light, and 28 frames without)

![button-scan-busyf14_disabled_full](https://user-images.githubusercontent.com/12900583/28852116-1db89a8a-7727-11e7-843b-5647901425bd.png)

**Comment by Sami32** [comment]
Wouaah, i would have ever imagined that.
That make me feel the need to buy a bigger screen :stuck_out_tongue_winking_eye: 

Damn, it is really nice !
Now i understand better the expression "see the world through the wrong end of his binoculars" :rofl:

**Comment by Nadahar** [comment]
@Sami32 Nomatter how big your screen was you wouldn't see all the details, the above images are in a much higher resolution. I know that I'm a bit mad, I just have such a hard time ignoring the details. My avatar is no better, this is how it really looks:

![nocrossallowed4_f90_1024_comp](https://user-images.githubusercontent.com/12900583/28852886-cff85096-772c-11e7-93ef-bb0ac7ef453e.png)

Perhaps the grimmes example of them all is the time I spent on a lot of details *inside* the RJ-45 socket. They aren't visible whatever resolution is used, since they are hidden by other objects. Still, I've spent time on micrometers inside it, rounding edges etc. Most of this is invisible also in these images, but I'm sure you get the idea:

![icon-status-disconnected_full_02](https://user-images.githubusercontent.com/12900583/28853010-ba65d464-772d-11e7-97b7-4cd8f8cff614.png)

![icon-status-connected_full_38](https://user-images.githubusercontent.com/12900583/28853079-411a5fe8-772e-11e7-954d-d3766d5f8080.png)

![icon-status-connected_full_91](https://user-images.githubusercontent.com/12900583/28853091-5f73bb92-772e-11e7-8257-21a8d7478dc1.png)

**Comment by Sami32** [comment]
I did already know that you was mad, but a genius one LOL

Now that i saw your avatar in big, i see that the red in the snake mouth is a little too "flashy" and i'm not sure about the red in its eye. As you're perfectionist i don't think it's a mistake, so the little blank point in the upper left should perhaps be a star ?
:smile:

**Comment by Nadahar** [comment]
@Sami32 There's no red in the snakes eye, that must be your screen filter. But, since you mention it, I did spend considerable time trying to make "real looking" skin inside the mouth (wet looking etc.), but I didn't really get it right and in the end concluded that what I had would have to do.

**Comment by Sami32** [comment]
Right, it was my red filter 8-/
Not for the white point in the red circle though ;-)
I'm just teasing you, it's an amazing work that require more patience that i'll never have.

**Comment by Nadahar** [comment]
There's a thunderstorm closing quickly here now, so I'm shutting down all my computers and will probably be offline soon.

**Comment by Nadahar** [comment]
@Sami32 I don't remember exactly, but I seem to remember that you said you had to use an older version of Firefox for some reason. For me it has become very slow after around version 50.

I finally got enough of this and found the solution, so if that's the problem you're having as well it might solve the problem. The cause is that Firefox "isolates" the GUI from the rest using two different processes, for "security reasons". For me this just kills the performance: 
* Open ```about:config```
* Search for ```autos```
* Find the option ```browser.tab.remote.autostart.2``` and set it to ```false```. This makes Firefox run as one process again, like it always use to before and like most other programs do. I really hate when security paranoia makes things unusable.

**Comment by Sami32** [comment]
It's much more simple than that in my case, after the version 48, Firefox doesn't support CPU without SSE2 anymore :wink: 

We cannot spit on Mozilla to care about our privacy and security using Sandbox.
I'll try it anyway. Thank you for sharing :smile:

**Comment by Nadahar** [comment]
@Sami32 I think that option became default after version 48. I'm just saying that the security measures shouldn't be applied without asking us when they have significant performance impact. I was almost ready to ditch Firefox because of this; what good is it to make at sandbox when people have to find other software? It sure doesn't make things more secure..

**Comment by Sami32** [comment]
Since most of the people don't really care about their own security, or are not willing to make the effort, i can understand their moves.
The people that do find the way to disable it.
Though i can dislike when someone do the choice for me thinking it's in my own interest ;-)

**Comment by Nadahar** [comment]
@Sami32 This wasn't easy to find because they don't inform you that this will make Firefox crawl to a stop when using many tabs. On the contrary, they say that "it might be slower in some cases but for most users there's no difference". Bullcrap, I have the same problem on all my computers running later versions of Firefox. If they presented me with a choice, I'd appreciate it, but doing it "the aPple way" is very bad and honestly I'm afraid this has already made them loose considerably marked shared.

A lot of people hate many things about Chrome, I've seen it described as a "spyware that can also display we pages", but: I've seen so many say that they use it because it's faster than Firefox.

In any case, that's Mozilla's problem, not ours, but I think they are making a big mistake.

When it comes to your CPU, even if you want to keep using ancient hardware you really *need* to get SSE2. It's become something that's expected by default by a lot of software. If you're worried about security, it should really worry you that you have to keep using old software (with security holes that has later become public knowledge that can be exploited by anyone). 

I thought maybe you could find a cheap, better CPU used, like the 3200+ (which must be a lot faster than the one you have). It turns out, it doesn't have SSE2 either, and it looks like none of the CPU's with socket A supports SSE2. That's a problem, because that means that your motherboard essentially does not support SSE2, so the problem isn't only the CPU.

Thus, this was another dead end. I've found one positive think though: The old AMD CPU's (like yours) doesn't have overheat protection (like almost any other CPU I've heard off) - which means they will just keep running when they become to how and basically fry themselves. All I have to do now is to trick you into running something very CPU intensive for a long time, and you can throw that CPU away without feeling bad 😈

**Comment by Sami32** [comment]
If you whish play FLAC into Firefox use the version 51. That's said, i share that the recent Firefox move is not my taste, and it's fine for me to stick to the version 48. I use TOR and I2P if i need to access some a little more private things.
If i really bother for security, i'll use only a secure Linux distro, like [Qube-OS](https://www.qubes-os.org/) or [TAILS](https://tails.boum.org/) :wink: On a fast USB 3.0 you can use it near everywhere while travelling.
I really love the work of [this guy](https://www.plop.at/en/bootmgr-new.html).
Absolutely ot Android or Chrome or Windows...

**Comment by Nadahar** [comment]
I'd like to see the day you manage to plug anything USB 3.0 into your achient computer 🤣 

The bootmanager looks nice from his description, but I really don't need it. I prefer virtual machines, where you keep things completely separated and errors in one system doesn't impact another 😉

**Comment by Sami32** [comment]
Na, not for mine ;-)
VM use CPU and lot of RAM, and when you travel around the world using public computer USB 3.0 is definitively the choice, IMO.
I believe separation being much better when booting directly on a USB key than on a VM.

**Comment by Nadahar** [comment]
I don't think separation is better, quite the contrary. When booting from USB you still have accesses the hardware and disks of the "host" computer. When using (proper) virtualization that isn't the case, all hardware is virtualized (simulated) and there's no way for the host and the guest to each other unless you specifically install/configure some communication channel. They can obviously communicate via the network, but that's on the same level as a computer can with any other resources on the local network.

In any case, if you plan to travel using public computers, using VM's isn't really an option. Installing and configuring a virtualization host isn't exactly quick, and you wouldn't be allowed to do that on a public computer either. The question is if you'd be able to boot from USB at all, but I guess that's where the "special" boot loader comes in 😄 

I wasn't talking about when travelling though, but about when I have the need to run different OS'es. I really love how easy it is using VMs when the host is installed and configured. And, they are completely isolated, I can copy VM's to easily duplicate a computer and I can take snapshots of it's state before trying something that I don't know if will work, allowing me to roll it back to the snapshot if things go wrong. That means rolling back everything, even what's in the RAM, so the VM is exactly as it was before you started.

**Comment by Sami32** [comment]
I was only thinking about travelling use.
> they are completely isolated

Well, i don't think so. Their drivers access is the main weakness, as if they get corrupt things can go wrong. At least it's how i understand it.

For my little use, or family one, i install [Deep Freeze](http://www.faronics.com/products/deep-freeze/), and no need to worry about virus or like in 99% of the cases.

**Comment by Nadahar** [comment]
@Sami32 There are many different virtualization solutions, but I use mostly ESXi - and all the VM's use virtual drivers for the hardware. The virtual drivers talk to virtual hardware and is as such isolated. The drivers for the host is especially built for this use, and the problem is that there are quite few drivers available for ESXi (as it's non-standard and VMWare mostly cares about making drivers for high-end equipment). To get it working properly, one has to be very specific in which hardware to use to make sure it there are ESXi drivers available for them. Obviously, the VM's will have problems if the "host" drivers doesn't work, but I think they are pretty well isolated.

**Comment by Sami32** [comment]
You still have white trailing space issue into you latest PR, and as that time it's a new one, it seem that you're not using anymore the "white commit".

**Comment by Sami32** [comment]
Why do we have in the code ?
```
<meta name="google-site-verification" content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU">
<meta name="google-site-verification" content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA">
<meta name="google-analytics" content="UA-3769691-2">
```

**Comment by Nadahar** [comment]
@Sami32 Where? It sounds like some ad identification stuff, but I can't find it when searching the code base.

**Comment by Sami32** [comment]
It was i thought, but the strange thing is that i'm able to refing it as well. I contain more than that, more personal stuff as you can see. Download it, i will delete it after.
If i remember correctly it was inside an `.icns` file into the resources folder, when i tried
[logo.icns.txt](https://github.com/DigitalMediaServer/DigitalMediaServer/files/1209004/logo.icns.txt)
 to understand what was this format.

**Comment by Sami32** [comment]
Now i remember, it happened when i uploaded the `logo.icns` from Github, taken from the resources folder.
Maybe i did something wrong.

**Comment by Sami32** [comment]
Now i think that happened when i first saved locally the `logo.icns` for after upload it in an other repository.
So, i certainly did a mistake at downloading that Mac icon from Github.

Sorry for the inconvenience :-/

**Comment by Sami32** [comment]
Sorry, i cannot remember where and what it was about.

I think that yesterday you asked me to look at an issue on the forum.
Could you point me again to it ? :flushed:

**Comment by Sami32** [comment]
It is just a question, a wonder, as i'm not sure about your intention, that's they could changes...

As you are working on engines registration, i think it's time for me to ask you about Handbrake and MEncoder.
From my side, MEncoder could be restricted to only DVD, as all the others task still link to it can be defered to MEncoder, so Hanbrake could do all the "dirty" job very properly (and it is also very very up to date concerning GPU optimisation) and defer it to FFmpeg.

What is your point of view, idea on that point ?

**Comment by Sami32** [comment]
Btw, `-tune zerolatency` shouldn't have been set as default, if set at all.
It is for real time streaming; i suspect that Jumpy was using it on some way, and the Web internet (_WEB.conf_) that near no one use as it's buggy and limited.
I read that could also make 2x slower the transcoding ?

Anyways, i think that `-tune fastdecode` or `-x264-params force-cfr=1` or even `-map_metadata -1 -map_chapters -1` should much more appropriate by default.

That's said, i don't share the `-level 3.1`, it should have been only added to the few renderers that need it, like some phones and old DVD players. Outpassing format specifications and limiting the quality with these 2 options was not a good choice, IMO.

I didn't even asked you to test `-preset ultrafast`, as lot of quality is lost, but we win 2x more speed, that could have made you able to stream 400 Mb/s video bitrate, but i suspect that you'll have complained about the video quality ? :wink:

**Comment by Nadahar** [comment]
@Sami32 My idea about how to handle the engines isn't fully formed yet, but I'm thinking of doing it quite differently in the end. I want to make classes whose responsibility is the discover the capabilities of each transcoding engine (I've already made basic tests in this PR which can easily be extended to check all kind of things). I then want to make other classes that determines what capabilities that is needed for a certain transcode. Using the enabled engines and comparing what's available and the requirements DMS can then figure out which transcoding profiles can be offered for a given media file. If multiple (enabled) transcoding engines meet all the requirements, the one with the highest priority will be chosen. If none of the enabled/available transcoding engines can meet the requirements for that profile, it won't be made available. If no transcoding engine meet the requirements for any valid profile, transcoding won't be offered.

I see no need to remove MEncoder, there's still some things that only MEncoder can do and it does work reasonably well in many cases. I do intend to give it a low default priority though.

I'm planning to support Handbrake and Libav as well. Even though I know that Libav offers less functionality than FFmpeg, I expect that what it does support it does with less bugs. I'd want Libav to be very high on the default priority, so that FFmpeg is only used when something that Libav doesn't support is required.

**Comment by Nadahar** [comment]
@Sami32 As you know, I haven't spent time studying the FFmpeg options, so I don't really know what most of them do. Some of them are kind of descriptive in their names, but I don't like guessing so I try not to assume too much without checking. I suspect that the description of many options can be very limited, so that it can be hard to figure out. In any case, I don't plan on looking at that before I get there and know what I'm looking for.

**Comment by Sami32** [comment]
That look a great idea to me, more efficient.
>  there's still some things that only MEncoder can do

Can be more precise ? because i don't see about what you're talking.
Libav ? well, i didn't thought about that as i'm not even sure if it will still exist or be maintened in 2 years...

I was just sharing my view on this, as i already read about, tried and tested, and used testers, many of them, and wishing know what you have in mind.

**Comment by Nadahar** [comment]
@Sami32 Handbrake use Libav as far as I know. I don't know the future of Libav, but I certainly like their philosophy better. If Libav is terminated, FFmpeg will loose a lot of its quality code as well as far as I understand, as FFmpeg merges "everything" from Libav. I suspect that the code dealing with Libav and the code dealing with FFmpeg will be very similar, so much can be reused. I actually hope the current situation stays as it is, where you have a "quality" lib you can use for common tasks and you have the possibility to fall back to less stable code when special stuff is needed. It sound like a reasonable choice to me.

If you look at Handbrake, they use MPlayer's libs for DVD parsing. I don't have enough information to tell if eveything MEncoder does can be replaces by something else, but I'm in no hurry to drop it.

**Comment by Sami32** [comment]
OK, i didn't checked what Handbrake use. I'll check later.
My view is [slighty different](https://lwn.net/Articles/650816/), but if they can find some sponsors they can perdure.

Btw, sorry to your wife, i wasn't intending to push anything in PS4_pro branch, i just did it by mistake when thinking pushing on my fork...

**Comment by Nadahar** [comment]
@Sami32 I don't quite see the difference between that said in the linked article and what I'm saying. Much of the "heavy lifting" is done by Libav, and FFmpeg is such a mess that basically only one person knows enough to figure it out. It would be a disaster if Libav were to die.

The "security issues" approach of investigating and doing "deeper" fixes is an approach I support, and the way we use it I'm pretty sure that most of the "security bugs" found is completely irrelevant anyway.

**Comment by Sami32** [comment]
> FFmpeg will loose a lot of its quality code as well as far as I understand, as FFmpeg merges "everything" from Libav

It was related to your words, as you can see around 30% Libav commits come from FFmpeg developpers, i don't share your "everything" view.
If Handbrake use it, it good for them, as without sponsors their future could be compromise, and the latest big one move to FFmpeg.
That was all what i tried to say.

**Comment by Nadahar** [comment]
@Sami32 You should know that you can't count contributions in commits. Look at many of my commits which does a lot compared to commits that only changes a few lines. I don't know enough the FFmpeg/Libav code and commit history to know what is what, but I do know that the "patchy", messy style of coding tends to generate much more commits than the more though-through style. This because the thought-through style tends to look deeper when addressing an issue, which leads to fewer but often larger commits. The "patchy" style first commit a non thought-through change and then discover that something else must also be changed (often by feedback), makes that commit, again only changing the obvious. This cycle repeats until nothing more is found or makes it through to the developer, or all the mess committed has made it impossible to get any further on the issue, where it tends to stay in its semi-broken state.

I never said that the Libav commits constituted "everything" that was committed to FFmpeg, I said that form what I understand more or less "everything" that is committed to Libav is also merged into FFmpeg. This is confirmed by the article you linked to, where you can see that the commits found in Libav is also found in FFmpeg.

I'm not saying that Libav is fantastic and FFmpeg is bad. FFmpeg does a lot of good stuff, but I believe that it's being run in a way that is a recipe for disaster in the long run. When the lead developer grows tired or can't continue for some reason, it's very likely to end up like MPlayer. I prefer quality over quantity ever day of the week as you know, but at the same time we need some of the stuff that's only available in FFmpeg. I dislike the "highjack" Libav attempted to do when the fork was made. It was more of a mutiny than a fork, and I think that was a big mistake. I value honesty, and users should be offered informed choices. There should have been a clear name and logo change from the start, and people should have the possibility to choose. There shouldn't be a need for a "conflict" between the two, the world should be tolerant enough to accept different approaches and philosopies.

That said, Libav is trying to be a "democratic" project while FFmpeg is to a large degree a "totalitarian" project. I know where I would prefer to make contributions. I really hope Libav doesn't die, and I believe its approach makes it much easier for new contributors to replace those withdrawing, giving it much better long-term survivability and sustainability.

**Comment by Sami32** [comment]
Don't misunderstand me, i didn't say that Libav is not good, just that i don't know about their future development and i guess that matter when we use it in a project, that said, nothing is sure on this earth...

Yes, big refactory can clean the mess when it exist, but also create issues that will need to be solved later. No solution is perfect, but it's the price to paid to avoid eternal patching until no one have an clear and full overview of the problem. Nothing personal here, i just say this to avoid misunderstanding appear.

**Comment by Nadahar** [comment]
@Sami32 As I said before, I expect that the arguments for FFmpeg and Libav are so similar that supporting both isn't much extra work.

Regarding big refactoring I don't think anybody wants that, and it obviously risks breaking things or creating new bugs. In my view, coding "quality" code is about trying to make things robust and versatile enough to avoid having to do big refactorings. But, when the code is already in a "beyond repair" state witch crucial shortcomings in structure and/or logic, I often see no other option to try to get it "under control".

**Comment by Sami32** [comment]
I'm not sure that Libav will be really usefull in our case, and also that it make the build again bigger.
Anyway, since many unuseful _flac.exe_ or _audio HD engines_ stay there when they should have been handled by FFmpeg, IMO, it doesn't really matter.

Whatever, it was just wonderings about your project. I project to get rid of them later on my fork.

Yeaah, but you are not "anybody" :wink: 
It should be a lovely and healthy preject, but i suspect that as soon as you will not maintain it yourself  the code will get "crowded" in the following years.

**Comment by Nadahar** [comment]
@Sami32 Much of the point with #2 is to "detach" DMS from the transcoding engines, so that people can use what they want or have. I'm not sure what should be bundled and what shouldn't be in the end, but by making DMS able to "detect" and use the engines that's available, the need to bundle is reduced. I guess there should be something bundled for those that doesn't want to understand the implications of choosing which engines to use, but I'm not finished figuring this out. My goal for now is the make the "link" weaker.

**Comment by Sami32** [comment]
Oh i see, at least i begin to see what you are targetting with this PR. That's a smart move, IMO, but i suspect that you could get headeaches while implementing it for the different Linux savor; i guess that limiting it to the main Linux savor should be suffisant ?

**Comment by Nadahar** [comment]
@Sami32 #2 is more a "framework" where the exact methods of detection can be customized for certain planforms/engines later. As for now, it relies on the executable being available on the ```PATH``` for in most cases. I think that will work for most Linux installations and flavors, but it will be relatively easy to add extra logic for detection where needed later. There is also a configurable "custom" option, where the user can specify the exact path to the executable.

**Comment by Sami32** [comment]
It seem that your are the only one seing MEncoder as still usefull for other thing that DVD playback.

https://github.com/UniversalMediaServer/UniversalMediaServer/issues/372#issuecomment-239922659
>  SubJunk commented on 15 Aug 2016 
@Goten87  it's likely that the sync issues are when using MEncoder but now by  default we use FFmpeg for almost everything. There's not much we can do  about the sync with MEncoder since it's all but abandoned so I think we  can close this

Or see also https://github.com/UniversalMediaServer/UniversalMediaServer/issues/1107

**Comment by Sami32** [comment]
I don't think that MEcoder can produce fragmented MP4...not telling that it doesn't support B-Frames handling, AFAICU.

I checked at Handbrake and saw that it use LIBAV, so i don't think that LIBAV need to be added separetely ?
Though i didn't saw that they use MPlayer code ?

**Comment by Nadahar** [comment]
@Sami32 They use the DVD libraries from MPlayer.

**Comment by Sami32** [comment]
Well, i only found trace that it use `libdvdcss` library, so if you've a link to share... :wink:

**Comment by Nadahar** [comment]
@Sami32 I don't remember where I found the overview now, but if you look [here](https://github.com/HandBrake/HandBrake/tree/master/contrib) you can see ```libdvdnav``` and ```libdvdread``` that is from MPlayer.

**Comment by Sami32** [comment]
Thank you :smile: 
I didn't know that `libdvdread` and `libdvdnav` was part of MPlayer :flushed:

**Comment by Sami32** [comment]
MEncoder don't have any static builds.
It stop to upgrade to the latest tsMuxer version that handle more format and PGS subtitles and others.
http://www.universalmediaserver.com/forum/viewtopic.php?f=4&t=8352&p=27369&hilit=tsMuxeR#p27360

At least now you have got my point of view on MEncoder :stuck_out_tongue_winking_eye:
